### ISSUE TITLE
React `Button` should be renamed to `Link` and not use `aria-popup` attribute

### ISSUE THREAD

**Issue by lb-** [wagtail/wagtail#11581]
### Is your proposal related to a problem?

Our current React component `Button` is only ever used as a link (with some extra onClick/navigation handling), it only renders an `a` element and hence should be a `Link` component instead.

### Describe the solution you'd like

1. Rename `client/src/components/Button/Button.tsx` to be `client/src/components/Link/Link.tsx`.
2. Remove the `dialogTrigger` trigger prop, this appears to be never used. Double check it's not used before proceeding.
3. Remove the `aria-haspopup={dialogTrigger ? 'dialog' : undefined}` usage on the rendering, this is not accessible even it was used (as it can only be applied to role="button" or similar).
4. Remove the Eslint rule ignoring `'jsx-a11y/role-supports-aria-props': 'off',` as this component is the only case where this rule is violated.
5. Update unit tests, imports and other code that uses the `Button`; `client/src/components/PageExplorer/PageExplorerHeader.tsx` & `client/src/components/PageExplorer/PageExplorerItem.tsx`

### Describe alternatives you've considered

* We could rename this to `NavigateLink` to be a bit more explicit about its ability to handle the page explorer navigation functionality, but this is probably fine to be `Link`.
* We could remove the component all together, it's only used in two files and those could be manually rendered as `a` link elements.

### Additional context

* This is not a current accessibility violation, but it's confusing for new developers and is a risk for building non-compliant React implementations.
* Generally, it's bad practice to make a component called `Button` always render an `a` and not a `button` element.
* This was written a long time ago and it's possible that the first implementation of this had some other usage (the dialog trigger) that is not required now. See 33af79a1c4c1203d270531176ba88c999cb23f87

### Working on this

* Anyone can contribute to this, you will need some knowledge of React, an ability to run all unit tests, linting and JS code rebuilding locally.
* View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by advikkabra** [comment]
I have started working on this.

**Comment by advikkabra** [comment]
#11585 submitted a PR regarding this.

**Comment by lb-** [comment]
Closed via https://github.com/wagtail/wagtail/pull/11585

Thanks @laymonage  I forgot to add to the commit message.
